spec/models/tag_rule_spec.rb

```diff
@@ -2,10 +2,21 @@
 
 require 'spec_helper'
 
+# This is used to test non implemented methods
+class TestTagRule < TagRule; end
+
 RSpec.describe TagRule do
   describe "validations" do
     it "requires a enterprise" do
       expect(subject).to belong_to(:enterprise)
     end
   end
+
+  describe '#tags' do
+    subject(:rule) { TestTagRule.new }
+
```

Comment:
Defining a new class like this leaks into the the global namespace for all specs. The best practice I've seen so far is to use `Class.new`:

https://github.com/openfoodfoundation/openfoodnetwork/blob/adddee2c3c4ff13098d5c520de5c1b64e9fc6c37/spec/models/spree/gateway_spec.rb#L6-L19

So here you could do:
```rb
subject(:rule) { Class.new(TagRule).new }
```